contracts/strategy/TradeFintechStrategy.sol

```diff
@@ -0,0 +1,134 @@
+        managerAddress = _managerAddress;
+        depositPeriodEnds = block.timestamp + _depositPeriod;
+        lockPeriodEnds = depositPeriodEnds + _lockPeriod;
+    }
+
+    function _harvestAndReport() internal override returns (uint256 _totalAssets) {
+        if (!TokenizedStrategy.isShutdown()) {
+            // deposit any loose funds
+            uint256 looseAsset = asset.balanceOf(address(this));
+            if (looseAsset > 0) {
+                uint256 _amount = Math.min(looseAsset, availableDepositLimit(address(this)));
+                asset.transfer(managerAddress, _amount);
+                totalInvestedInRWA += _amount;
+            }
+        }
+        _totalAssets = totalInvestedInRWA + asset.balanceOf(address(this));
+    }
+
+    function availableDepositLimit(address /*_owner*/) public view override returns (uint256) {
+        // Return the remaining room.
+        return type(uint256).max - asset.balanceOf(address(this));
+    }
+
+    function availableWithdrawLimit(address /*_owner*/) public view override returns (uint256) {
+        return TokenizedStrategy.totalIdle() + totalInvestedInRWA;
+    }
+
+    function _deployFunds(uint256 _amount) internal override {
+        require(block.timestamp <= depositPeriodEnds, "Deposit period has ended");
+        asset.transfer(managerAddress, _amount);
+        totalInvestedInRWA += _amount;
+    }
+
+    function _freeFunds(uint256 _amount) internal override {
+        require(block.timestamp > lockPeriodEnds, "Lock period has not ended");
+        // We don't check available liquidity because we need the tx to
+        // revert if there is not enough liquidity so we don't improperly
+        // pass a loss on to the user withdrawing.
+        asset.safeTransferFrom(managerAddress, address(this), _amount);
+        totalInvestedInRWA -= _amount;
+    }
+
+    /**
+     * @dev Optional function for a strategist to override that will
+     * allow management to manually withdraw deployed funds from the
+     * yield source if a strategy is shutdown.
+     *
+     * This should attempt to free `_amount`, noting that `_amount` may
+     * be more than is currently deployed.
+     *
+     * NOTE: This will not realize any profits or losses. A separate
+     * {report} will be needed in order to record any profit/loss. If
+     * a report may need to be called after a shutdown it is important
+     * to check if the strategy is shutdown during {_harvestAndReport}
+     * so that it does not simply re-deploy all funds that had been freed.
+     *
+     * EX:
+     *   if(freeAsset > 0 && !TokenizedStrategy.isShutdown()) {
+     *       depositFunds...
+     *    }
+     *
+     * @param _amount The amount of asset to attempt to free.
+     */
+    function _emergencyWithdraw(uint256 _amount) internal override {
+        uint256 amountToWithdraw = Math.min(_amount, totalInvestedInRWA);
```

Comment:
What happens with non invested assets? Just asking atm

Reply:
What's happening here on the _emergencyWithdraw function right now is:

1. Strategy needs to be shutdown before
2. The minimum amount between what is invested (the amount that was transferred from the Strategy to the Manager) and the requested amount will be the amountToWithdraw
3. Strategy tries to transfer the amountToWithdraw from the Manager to itself
4. Strategy tries to send the amountToWithdraw to the address who is the management of the Strategy (usually who deployed it. The name is confusing here)

Non invested assets will stay in the Strategy as idle funds. Will fix this part to have it also transferred.

Reply:
Yes, I was curious if there are some funds left